### PR TITLE
feat: mock secureboot enabled

### PR DESCRIPTION
**Pull request by Unix4ever**
Secureboot can be enabled through the provider data.

### CODE REVIEW

File: internal/pkg/provider/provision.go [hunk 1]
```diff
@@ -43,6 +47,13 @@ func (p *Provisioner) ProvisionSteps() []provision.Step[*resources.Machine] {
 				return fmt.Errorf("failed to get the machine task: %w", err)
 			}
 
+			var pd providerData
+
+			err = pctx.UnmarshalProviderData(&pd)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal provider data")
```
Comment (DmitriyMV) [inline internal/pkg/provider/provision.go:54]: Forgot the error `...%w:" err`?

**Review by DmitriyMV — APPROVED** [review]

**Review by smira — APPROVED** [review]
should it validate something, i.e. the installer image reference?

**Comment by Unix4ever** [comment]
> should it validate something, i.e. the installer image reference?

Yeah, I had an idea to have `-secureboot` suffix validation, but wasn't sure about that as the plan is to drop that.

File: internal/pkg/provider/provision.go (hunk 1 above)
Comment (Unix4ever) [reply internal/pkg/provider/provision.go:54]: yep, thanks

**Pull request merged**